packages/moonstone/Dropdown/DropdownList.js

```diff
@@ -0,0 +1,193 @@
+import kind from '@enact/core/kind';
+import hoc from '@enact/core/hoc';
+import EnactPropTypes from '@enact/core/internal/prop-types';
+import Spotlight from '@enact/spotlight';
+import ri from '@enact/ui/resolution';
+import PropTypes from 'prop-types';
+import compose from 'ramda/src/compose';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import Item from '../Item';
+import Skinnable from '../Skinnable';
+import VirtualList from '../VirtualList';
+
+import css from './Dropdown.module.less';
+
+const scrollOffset = 2;
+
+const isSelectedValid = ({children, selected}) => Array.isArray(children) && children[selected] != null;
+
+const DropdownListBase = kind({
+	name: 'DropdownListBase',
+
+	propTypes: {
+		/*
+		* The selections for Dropdown
+		*
+		* @type {String[]|Array.<{key: (Number|String), children: (String|Component)}>}
+		*/
+		children: PropTypes.oneOfType([
+			PropTypes.arrayOf(PropTypes.string),
+			PropTypes.arrayOf(PropTypes.shape({
+				children: EnactPropTypes.renderable.isRequired,
+				key: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired
+			}))
+		]),
+
+		/*
+		* Called when an item is selected.
+		*
+		* @type {Function}
+		*/
+		onSelect: PropTypes.func,
```

Comment:
Required?  We don't validate presence below.

Reply:
I'll add a guard.